cpp/include/cudf/quantiles.hpp

```diff
@@ -94,5 +95,31 @@ std::unique_ptr<table> quantiles(
   std::vector<null_order> const& null_precedence = {},
   rmm::mr::device_memory_resource* mr            = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Calculate approximate percentiles on an input tdigest column.
+ *
+ * tdigest (https://arxiv.org/pdf/1902.04023.pdf) columns are produced specifically
+ * by the TDIGEST and MERGE_TDIGEST aggregations.  These columns represent
+ * compressed representations of a very large input data set that can be
+ * queried for quantile information.
+ *
+ * Produces a LIST column where each row `i` represents output from querying the
+ * corresponding tdigest from `input` row `i`. The length of each output list
+ * is the number of percentages specified in `percentages`.
+ *
+ * @param input           tdigest input data. One tdigest per row.
+ * @param percentiles     Desired percentiles in range [0, 1].
+ * @param output_type     Desired (numeric) output type.
+ * @param mr              Device memory resource used to allocate the returned column's device
+ * memory
+ *
+ * @returns LIST Column containing requested percentile values.
+ */
+std::unique_ptr<column> percentile_approx(
+  structs_column_view const& input,
+  column_view const& percentiles,
+  cudf::data_type output_type         = data_type{type_id::FLOAT64},
```

Comment:
This is suspicious. We almost never allow controlling the output data type (see `quantiles` API above). This almost always necessitates a double-dispatch (bad for compile time/binary size).

Why is this needed here?

Reply:
I see, we can avoid the double dispatch because the type(s) of the tdigest column are known and don't need dispatch.

I still don't like this due to the inconsistency with pretty much every other function in libcudf.

Reply:
Correct. This is a little bit of a bonus feature for Spark.  The idea is that you might be computing the percentiles on some big pile of say, decimals.   And at the end of the day, you want a decimal value, not a double.  I suppose in theory this could be handled by the plugin?   @andygrove  @mythrocks

Reply:
> Those disappear into the tdigests after the aggregation.

Is that the right behavior? When computing a tdigest for a set of fixed-point values, shouldn't the centroid value_type be fixed-point? If that were the case, then you wouldn't need to have the knob to control the output type.

This is how something like groupby mean ([should](https://github.com/rapidsai/cudf/issues/9224)) works. The result type for computing the mean on fixed-point would also be fixed-point.

Reply:
The way it seems to be done in general is to keep it as all doubles internally.   In fact, even in Spark's CPU-side implementation (which uses a completely different type of digest) uses doubles all the way.   In either case, we removed the ability to specify the output type as the plugin can work around it.